src/main/java/no/unit/nva/hamcrest/DoesNotHaveEmptyValues.java

```diff
@@ -40,22 +46,42 @@ public DoesNotHaveEmptyValues() {
      * @param <R>        the type of the object in test.
      * @return a matcher.
      */
-    public static <R> DoesNotHaveEmptyValues<R> doesNotHaveEmptyValuesIgnoringClasses(List<Class<?>> ignoreList) {
+    public static <R> DoesNotHaveEmptyValues<R> doesNotHaveEmptyValuesIgnoringClasses(Set<Class<?>> ignoreList) {
         DoesNotHaveEmptyValues<R> matcher = new DoesNotHaveEmptyValues<>();
-        ArrayList<Class<?>> newStopRecursionClasses = new ArrayList<>();
+        Set<Class<?>> newStopRecursionClasses = new HashSet<>();
         newStopRecursionClasses.addAll(matcher.stopRecursionClasses);
         newStopRecursionClasses.addAll(ignoreList);
         matcher.stopRecursionClasses = newStopRecursionClasses;
         return matcher;
     }
 
+    public static <R> DoesNotHaveEmptyValues<R> doesNotHaveEmptyValuesIgnoringFields(Set<String> ignoreList) {
+        DoesNotHaveEmptyValues<R> matcher = new DoesNotHaveEmptyValues<>();
+        matcher.ignoreFields = addFieldPathDelimiterToRootField(ignoreList);
+        return matcher;
+    }
+
+    private static Set<String> addFieldPathDelimiterToRootField(Set<String> ignoreList) {
```

Comment:
fix for the fact that the fields of the root are represented as ".fieldName". I will create a better fix when there is time